### PR TITLE
[#16] [Integrate] As a logged-in user, I can pull down to refresh the survey list

### LINKED ISSUE #16: [Integrate] As a logged-in user, I can pull down to refresh the survey list
### Why
On the Home screen, the user can see the latest surveys.

### Acceptance Criteria
- On the Home screen, when the user pulls down, the app will fetch the data, discard the existing surveys and display the latest surveys.
- Unit tests are added and passed.

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
- Closes #16 

## What happened 👀

On the Home screen, the user can see the latest surveys.

## Insight 📝

- On the Home screen, when the user pulls down, the app will fetch the data, discard the existing surveys and display the latest surveys.
- Unit tests are added and passed.

## Proof Of Work 📹

Show us the implementation: screenshots, gif, etc.

https://github.com/nimblehq/flutter-ic-kaung-thieu/assets/32578035/6557f8d7-636f-4fb6-a9f9-c52eb1e41446

### CODE REVIEW

File: lib/usecases/clear_cached_surveys_use_case.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+final clearCachedSurveysUseCaseProvider = Provider(
+    (ref) => ClearCachedSurveysUseCase(ref.watch(hiveStorageProvider)));
+
+class ClearCachedSurveysUseCase extends NoParamsUseCase<int> {
+  final HiveStorage _hiveStorage;
+
+  ClearCachedSurveysUseCase(this._hiveStorage);
+
+  @override
+  Future<Result<int>> call() async {
```
Comment (manh-t) [inline lib/usecases/clear_cached_surveys_use_case.dart:14]: we don't use the result so it's better to return `void`
```suggestion
class ClearCachedSurveysUseCase extends NoParamsUseCase<void> {
  final HiveStorage _hiveStorage;

  ClearCachedSurveysUseCase(this._hiveStorage);

  @override
  Future<Result<void>> call() async {
```

File: lib/usecases/clear_cached_surveys_use_case.dart [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    final result = await _hiveStorage.clearSurveys();
+    return Success(result);
```
Comment (manh-t) [inline lib/usecases/clear_cached_surveys_use_case.dart:16]:
```suggestion
    return Success(null);
```

File: test/screens/home/home_view_model_test.dart [hunk 3]
```diff
@@ -128,5 +133,20 @@ void main() {
         mockGetSurveysUseCase.call(any),
       ]);
     });
+
+    test('When call refresh, it call getCachedSurveysUseCase', () async {
```
Comment (manh-t) [inline test/screens/home/home_view_model_test.dart:137]: Please be more careful with grammar issues 🙏 
```suggestion
    test('When calling refresh method, it calls getCachedSurveysUseCase', () async {
```

File: lib/usecases/clear_cached_surveys_use_case.dart (hunk 1 above)
Comment (kaungkhantsoe) [reply lib/usecases/clear_cached_surveys_use_case.dart:14]: @manh-t I tried to return `void` at first. But it cannot be unit tested. I researched a little and I found one website that says we should avoid returning `void` as much as we could.

**Review by manh-t — APPROVED** [review]

**Review by manh-t — APPROVED** [review]

**Review by ducbm051291 — APPROVED** [review]

**Pull request merged**